src/Lib.P2P.Tests/Multiplex/MuxerTest.cs

```diff
@@ -0,0 +1,189 @@
+            var channel = new MemoryStream();
+            var muxer1 = new Muxer {Channel = channel, Initiator = true};
+            await muxer1.CreateStreamAsync("foo");
+            await muxer1.CreateStreamAsync("bar");
+
+            channel.Position = 0;
+            var muxer2 = new Muxer {Channel = channel};
+            var n = 0;
+            muxer2.SubstreamCreated += (s, e) => ++n;
+            await muxer2.ProcessRequestsAsync();
+            Assert.AreEqual(2, n);
+        }
+
+        [TestMethod]
+        public async Task NewStream_AlreadyAssigned()
+        {
+            var channel = new MemoryStream();
+            var muxer1 = new Muxer {Channel = channel, Initiator = true};
+            await muxer1.CreateStreamAsync("foo");
+            var muxer2 = new Muxer {Channel = channel, Initiator = true};
+            await muxer2.CreateStreamAsync("bar");
+
+            channel.Position = 0;
+            var muxer3 = new Muxer {Channel = channel};
+            await muxer3.ProcessRequestsAsync(new CancellationTokenSource(500).Token);
+
+            // The channel is closed because of 2 new streams with same id.
+            Assert.IsFalse(channel.CanRead);
+            Assert.IsFalse(channel.CanWrite);
+        }
+
+        [TestMethod]
+        public async Task NewStream_Event()
+        {
+            var channel = new MemoryStream();
+            var muxer1 = new Muxer {Channel = channel, Initiator = true};
+            await muxer1.CreateStreamAsync("foo");
+            await muxer1.CreateStreamAsync("bar");
+
+            channel.Position = 0;
+            var muxer2 = new Muxer {Channel = channel};
+            var createCount = 0;
+            muxer2.SubstreamCreated += (s, e) => { ++createCount; };
+            await muxer2.ProcessRequestsAsync();
+            Assert.AreEqual(2, createCount);
+        }
+
+        [TestMethod]
+        public async Task CloseStream_Event()
+        {
+            var channel = new MemoryStream();
+            var muxer1 = new Muxer {Channel = channel, Initiator = true};
+            await using (await muxer1.CreateStreamAsync("foo"))
+            {
+                await using (await muxer1.CreateStreamAsync("bar"))
+                {
+                    // open and close a stream.
+                }
+            }
+
+            channel.Position = 0;
+            var muxer2 = new Muxer {Channel = channel};
+            var closeCount = 0;
+            muxer2.SubstreamClosed += (s, e) => { ++closeCount; };
+            await muxer2.ProcessRequestsAsync();
+            Assert.AreEqual(2, closeCount);
+        }
+
+        [TestMethod]
+        public async Task AcquireWrite()
+        {
+            var muxer = new Muxer();
+            var tasks = new List<Task<string>>
+            {
+                Task.Run(async () =>
+                {
+                    using (await muxer.AcquireWriteAccessAsync())
+                    {
+                        await Task.Delay(100);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)